multimodal-search-tirg/TirgMultimodalEncoder /__init__.py

```diff
@@ -52,25 +53,24 @@ def _get_features(self, data):
         if self.channel_axis != self._default_channel_axis:
             visual_data = np.moveaxis(visual_data, self.channel_axis, self._default_channel_axis)
         textual_data = data[(self.positional_modality.index('textual'))]
-        
-        visual_data = torch.from_numpy(visual_data.astype('float32'))
-        textual_data = torch.from_numpy(textual_data.astype('float32'))
-        
+
+        visual_data = torch.stack(visual_data).float()
+
         if self.on_gpu:
             visual_data = visual_data.cuda()
             textual_data = textual_data.cuda()
-            
+
         img_features = self.model.extract_img_feature(visual_data)
         text_features = self.model.extract_text_feature(textual_data)
-        
-        return self.model.compose_img_text_features(img_features, text_feature)
+
+        return self.model.compose_img_text_features(img_features, text_features)
 
     @batching
     @as_ndarray
-    def encode(self, *data: 'np.ndarray', args, **kwargs) -> 'np.ndarray':
+    def encode(self, data: list, *args, **kwargs) -> 'np.ndarray':
```

Comment:
this signature should be:

```
def encode(self, *data: 'np.ndarray', **kwargs) -> 'np.ndarray':
```

Reply:
the point is we added also `*args` which can't be done

Reply:
done